### PR TITLE
DOC: fix docs build CI

### LINKED ISSUE #312: Docs CI build fails
https://github.com/pcdshub/pytmc/actions/runs/5316719264/jobs/9626518358
```
Theme error:
An error happened in rendering the page getting_started.
Reason: UndefinedError("'style' is undefined")
make: *** [Makefile:20: html] Error 2
Error: Process completed with exit code 2.
```

### PR DESCRIPTION
**Pull request by ZLLentz**
closes #312 if the docs build CI passes

- pin sphinx to <7 to avoid an incompatibility
- make the structure of the docs dir the same as other libraries
- update some missing gitignore items

### CODE REVIEW

**Review by tangkong — APPROVED** [review]
Docs build, and that's basically all I need to see 👍

**Review by klauer — APPROVED** [review]
LGTM

**Comment by klauer** [comment]
Some unrelated docs things to fix at some point:
```
/home/runner/work/pytmc/pytmc/docs/source/release_notes.rst:46: ERROR: Unexpected indentation.
/home/runner/work/pytmc/pytmc/docs/source/release_notes.rst:49: WARNING: Block quote ends without a blank line; unexpected unindent.
```

**Comment by ZLLentz** [comment]
One more curveball: does the sphinx pin belong in ci helpers instead, since I think lots of repos will have this problem?

**Comment by tangkong** [comment]
What were the actual details of the sphinx error?  I assumed it was specific to this repo but I would believe it's more widespread.

**Comment by ZLLentz** [comment]
tl;dr there is no sphinx-rtd-theme released with full sphinx 7 compatibility

**Comment by ZLLentz** [comment]
closing and reopening to refresh the ci without adding a commit

**Comment by ZLLentz** [comment]
I think this is nearly in its final form
There's still a pin added to the requirements file, but it's a > pin for sphinx_rtd_theme to prevent the older build with no knowledge about sphinx pinning from being installed
This hopefully means that it will switch to sphinx 7 seamlessly when it is supported (or, alternatively it will break again)
Unfortunately I think this still means that every repo needs to add the same pin?

CI is still pointing to the other branch

File: .github/workflows/standard.yml [hunk 1]
```diff
@@ -9,7 +9,8 @@ on:
 
 jobs:
   standard:
-    uses: pcdshub/pcds-ci-helpers/.github/workflows/python-standard.yml@master
+    #uses: pcdshub/pcds-ci-helpers/.github/workflows/python-standard.yml@master
+    uses: pcdshub/pcds-ci-helpers/.github/workflows/python-standard.yml@ref_use_pip_extras
```
Comment (klauer) [inline .github/workflows/standard.yml:13]: Reminder to fix this before merging

**Comment by ZLLentz** [comment]
This seems like it may be fully done pending CI

**Comment by klauer** [comment]
Looks good to me 👍
